libraries/AP_HAL_ChibiOS/hwdef/ARK_FPV/hwdef.dat

```diff
@@ -0,0 +1,215 @@
+
+# Analog battery input scaling
+# See STM32H743xx.py
+define HAL_BATT_VOLT_PIN 9
+define HAL_BATT_CURR_PIN 12
+define HAL_BATT_VOLT_SCALE 21
+define HAL_BATT_CURR_SCALE 120
+define HAL_BATT_MONITOR_DEFAULT 4
+
+# SPI1 - IIM42653 IMU
+PA5 SPI1_SCK SPI1
+PB5 SPI1_MOSI SPI1
+PG9 SPI1_MISO SPI1
+PI9 IMU1_CS CS
+PF2 IMU1_DRDY INPUT
+
+# SPI6 - external (SPI OSD or IMU)
+PB3  SPI6_SCK SPI6
+PA6  SPI6_MISO SPI6
+PG14 SPI6_MOSI SPI6
+PI10 EXT1_CS CS
+PD11 DRDY_ADIS16507 INPUT GPIO(93)
+
+# GPIO(93) for data ready on ADIS16507
+# define ADIS_DRDY_PIN 93
+
+# Motors, see STM32H743xx.py
+PI0  TIM5_CH4 TIM5 PWM(1) GPIO(50)
+PH12 TIM5_CH3 TIM5 PWM(2) GPIO(51) BIDIR
+PH11 TIM5_CH2 TIM5 PWM(3) GPIO(52)
+PH10 TIM5_CH1 TIM5 PWM(4) GPIO(53) BIDIR
+PI5  TIM8_CH1 TIM8 PWM(5) GPIO(54) BIDIR
+PI6  TIM8_CH2 TIM8 PWM(6) GPIO(55)
+PI7  TIM8_CH3 TIM8 PWM(7) GPIO(56) BIDIR
+PI2  TIM8_CH4 TIM8 PWM(8) GPIO(57)
+PD12 TIM4_CH1 TIM4 PWM(9) GPIO(58)
+
+# PWM output for buzzer
+PF9 TIM14_CH1 TIM14 GPIO(77) ALARM
+
+# CAN bus
+PD0 CAN1_RX CAN1
+PD1 CAN1_TX CAN1
+
+# I2C1 (GPS1 compass, external)
+PB9 I2C1_SDA I2C1
+PB8 I2C1_SCL I2C1
+
+# I2C2 (BMP390 baro, internal)
+PF1 I2C2_SCL I2C2
+PF0 I2C2_SDA I2C2
+
+# I2C4 (IIS2MDC compass, internal)
+PF14 I2C4_SCL I2C4
+PF15 I2C4_SDA I2C4
+
+# order of I2C buses
+I2C_ORDER I2C1 I2C2 I2C4
+define HAL_I2C_INTERNAL_MASK 6
+
+# IMU heater
+PB10 HEATER_EN OUTPUT LOW GPIO(80)
+define HAL_HEATER_GPIO_PIN 80
+define HAL_HAVE_IMU_HEATER 1
+define HAL_IMU_TEMP_DEFAULT 45
+define HAL_IMUHEAT_P_DEFAULT 50
+define HAL_IMUHEAT_I_DEFAULT 0.07
+# TODO: should we use this?
+# define HAL_IMU_TEMP_MARGIN_LOW_DEFAULT 5
+
+# Armed indication
+PE6 nARMED OUTPUT HIGH
+
+# Power enable pins
+PC13 VDD_3V3_SD_CARD_EN OUTPUT HIGH
+PI11 VDD_3V3_SENSORS1_EN OUTPUT HIGH
+
+# 12V enable, start with it off
+PG4 nVDD_5V_PERIPH_EN OUTPUT HIGH
+
+# Power sensing
+PE15 VDD_5V_PERIPH_nOC INPUT PULLUP
+PF13 VDD_5V_HIPOWER_nOC INPUT PULLUP
+PG1 VDD_BRICK_nVALID INPUT PULLUP
+
+# MicroSD
+PD6  SDMMC2_CK SDMMC2
+PD7  SDMMC2_CMD SDMMC2
+PB14 SDMMC2_D0 SDMMC2
+PB15 SDMMC2_D1 SDMMC2
+PG11 SDMMC2_D2 SDMMC2
+PB4  SDMMC2_D3 SDMMC2
+define FATFS_HAL_DEVICE SDCD2
+
+# LEDs
+PE3 LED_RED OUTPUT OPENDRAIN GPIO(90) HIGH
+PE4 LED_GREEN OUTPUT OPENDRAIN GPIO(91) HIGH
+PE5 LED_BLUE OUTPUT OPENDRAIN GPIO(92) HIGH
+define AP_NOTIFY_GPIO_LED_RGB_RED_PIN 90
+define AP_NOTIFY_GPIO_LED_RGB_GREEN_PIN 91
+define AP_NOTIFY_GPIO_LED_RGB_BLUE_PIN 92
+define AP_NOTIFY_GPIO_LED_RGB_ENABLED 1
+
+# ID pins
+PG0 HW_VER_REV_DRIVE OUTPUT LOW
+PH4 HW_REV_SENS ADC3 SCALE(1)
+
+# Barometer on i2c_2
+BARO BMP388 I2C:1:0x76
+
+# Compass
+COMPASS IIS2MDC I2C:2:0x1E false ROTATION_NONE
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+
+# IIM42653 IMU on SPI1
+SPIDEV iim42653  SPI1 DEVID1  IMU1_CS MODE3  2*MHZ  8*MHZ
+IMU Invensensev3 SPI:iim42653 ROTATION_YAW_270
+
+# SPI6 external bus
+# SPIDEV adis16507 SPI6 DEVID1  EXT1_CS MODE3  1*MHZ  2*MHZ
+# IMU ADIS1647x    SPI:adis16507 ROTATION_NONE ADIS_DRDY_PIN
```

Comment:
Should I have this driver active by default? Along with any other IMU drivers we want to support on external SPI?

Reply:
yes, otherwise it won't probe for these